src/states/approved_peer/mod.rs

```diff
@@ -1161,34 +1858,110 @@ impl Elder {
+                    .filter_outgoing(msg, p2p_node.public_id())
+                    .is_new()
+            })
+            .map(|node| *node.peer_addr())
+            .collect();
+
+        let cheap_bytes_clone = msg.full_bytes().clone();
+        self.send_message_to_targets(&targets, dg_size, cheap_bytes_clone);
+
+        Ok(())
     }
 
-    // Signs and proves the given `RoutingMessage` and wraps it in `SignedRoutingMessage`.
-    fn to_accumulating_message(
-        &self,
-        dst: DstLocation,
-        variant: Variant,
-        node_knowledge_override: Option<u64>,
-    ) -> Result<AccumulatingMessage> {
-        let proof = self.chain.prove(&dst, node_knowledge_override);
-        let pk_set = self.our_section_bls_keys().clone();
-        let secret_key = self.chain.our_section_bls_secret_key_share()?;
+    ////////////////////////////////////////////////////////////////////////////
+    // Miscellaneous
+    ////////////////////////////////////////////////////////////////////////////
 
-        let content = PlainMessage {
-            src: *self.our_prefix(),
-            dst,
-            variant,
-        };
+    fn print_rt_size(&self) {
+        const TABLE_LVL: log::Level = log::Level::Info;
+        if log_enabled!(TABLE_LVL) {
+            let status_str = format!(
+                "{} - Routing Table size: {:3}",
+                self,
+                self.chain.elders().count()
+            );
+            let network_estimate = match self.chain.network_size_estimate() {
+                (n, true) => format!("Exact network size: {}", n),
+                (n, false) => format!("Estimated network size: {}", n),
+            };
+            let sep_len = cmp::max(status_str.len(), network_estimate.len());
+            let sep_str = iter::repeat('-').take(sep_len).collect::<String>();
+            log!(target: "routing_stats", TABLE_LVL, " -{}- ", sep_str);
+            log!(target: "routing_stats", TABLE_LVL, "| {:<1$} |", status_str, sep_len);
+            log!(target: "routing_stats", TABLE_LVL, "| {:<1$} |", network_estimate, sep_len);
+            log!(target: "routing_stats", TABLE_LVL, " -{}- ", sep_str);
+        }
+    }
 
-        AccumulatingMessage::new(content, secret_key, pk_set, proof)
+    // Ignore stale GenesisUpdates
+    fn is_genesis_update_new(&self, gen_pfx_info: &GenesisPfxInfo) -> bool {
+        gen_pfx_info.parsec_version > self.gen_pfx_info.parsec_version
     }
 
-    pub fn in_src_location(&self, src: &SrcLocation) -> bool {
-        self.chain.in_src_location(src)
+    // Connect to all elders from our section or neighbour sections that we are not yet connected
+    // to and disconnect from peers that are no longer elders of neighbour sections.
+    fn update_peer_connections(&mut self, change: &EldersChange) {
+        if !self.chain.split_in_progress() {
+            let our_needed_connections: HashSet<_> = self
+                .chain
+                .known_nodes()
```

Comment:
just to confirm this `known_nodes` return only elders ?
as the comment says `no longer elders of neighbour sections`

Reply:
No, `known_nodes` returns all nodes (elders or not) that we know of. So the comment is wrong. Will fix it.

Reply:
Ah, but we only know about *elders* of the neighbour sections, so it isn't wrong.

Reply:
hmm...  still worth to at least put a comment to clarify?